azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/util/BlobHelper.java

```
@@ -0,0 +1,100 @@
+      }
+      finally {
+         Closeables2.closeQuietly(azureBlob);
+      }
+   }
+
+   public static boolean customImageExists(String storage, String key) {
+      final AzureBlobClient azureBlob = ContextBuilder.newBuilder("azureblob")
+              .credentials(storage, key)
+              .buildApi(AzureBlobClient.class);
+
+      try {
+         return azureBlob.containerExists("system");
+      }
+      catch (Exception e) {
+         e.printStackTrace();
+         return false;
+      }
+      finally {
+         Closeables2.closeQuietly(azureBlob);
+      }
+   }
+
+   public static List<VMImage> getImages(String containerName, String group,
+                                         String storageAccountName, String key, String offer, String location) {
+      final AzureBlobClient azureBlob = ContextBuilder.newBuilder("azureblob")
+              .credentials(storageAccountName, key)
+              .buildApi(AzureBlobClient.class);
+
+
+      List<VMImage> list = new ArrayList<VMImage>();
+      try {
+         BoundedSet<ContainerProperties> containerList = azureBlob.listContainers();
+         for (ContainerProperties props : containerList) {
+            if (props.getName().equals("system")) {
+               ListBlobsResponse blobList = azureBlob.listBlobs("system");
+               String osDisk = "";
+               String dataDisk = "";
+
+               for (BlobProperties blob : blobList) {
+                  String name = blob.getName();
+
+                  if (dataDisk.length() == 0) dataDisk = name.substring(1 + name.lastIndexOf('/'));
+                  else if (osDisk.length() == 0) osDisk = name.substring(1 + name.lastIndexOf('/'));
+               }
+               final VMImage ref = VMImage.create(group, storageAccountName, osDisk, dataDisk, "test-create-image", "custom", location);
+               list.add(ref);
+            }
+         }
+      } catch (Exception e) {
+         e.printStackTrace();
+      }
```

Same here. Let the method fail on uncontrolled errors.